src/websocket_handle/ws_handle.go

```diff
@@ -0,0 +1,98 @@
+package websocket_handle
+
+import (
+	"log"
+	"net/http"
+	"sync"
+
+	"github.com/HyperloopUPV-H8/Backend-H8/websocket_handle/models"
+	"github.com/gorilla/mux"
+	"github.com/gorilla/websocket"
+)
+
+type WSHandle struct {
+	handles map[string]chan models.MessageTarget
+	conns   map[string]*websocket.Conn
+	connsMx sync.Mutex
+}
+
+func RunWSHandle(router *mux.Router, route string, handles map[string]chan models.MessageTarget) *WSHandle {
+	handle := &WSHandle{
+		handles: handles,
+		conns:   make(map[string]*websocket.Conn),
+		connsMx: sync.Mutex{},
+	}
+
+	router.HandleFunc(route, handle.handleConn)
+
+	go handle.handleTx()
+
+	return handle
+}
+
+func (handle *WSHandle) handleTx() {
+	for {
+		for _, output := range handle.handles {
+			select {
+			case msg := <-output:
+				if len(msg.Target) == 0 {
+					for _, conn := range handle.conns {
+						err := conn.WriteJSON(msg.Msg)
+						if err != nil {
+							handle.Close(conn)
+						}
+					}
+				} else {
+					for _, target := range msg.Target {
```

Comment:
What does the message target represent?

Reply:
Where did the message come from or for who the message is, this allows to send the same message to multiple clients or to answer a specific request